test/integration/job/job_test.go

```diff
@@ -1026,6 +1026,107 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+func TestScalableIndexedJob(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.ScalableIndexedJob, true)()
+
+	closeFn, restConfig, clientSet, ns := setup(t, "indexed")
```

Comment:
Lets try to create a test table to cover the following scenarios:
1) Feature turned off, mutating disallowed
2) Scale up, verify that the range expands, and the job finishes successfully when indexes including the ones in the new range exit successfully.
3) Scale down, verify that the range shrinks, and the job finishes successfully when indexes only in the smaller range finishes successfully
4) Scale down, verify that failures that happened for indexes that are now outside the range still count against the backofflimit
5) Index finishes successfully, scale down to exclude the index, then scale up to include it back. Verify that the index was restarted
6) Scale down to 0, verify the job finishes successfully.

Reply:
Ok makes sense, I will work on these test scenarios and let you know when the PR is ready for another look

Reply:
I created a table driven test but in order to test all these scenarios it became quite large/complicated. All the tests are passing but I may refactor it into multiple tests, what do you think?